pysmt/test/test_bdd.py

```diff
@@ -115,6 +122,41 @@ def test_quantifier_eliminator(self):
         g = qelim(f, solver_name="bdd")
         self.assertEqual(g, self.y)
 
+    @skipIfSolverNotAvailable("bdd")
+    def test_reordering(self):
+        from pysmt.solvers.bdd import BddSolver
+        with BddSolver(get_env(), pysmt.logics.BOOL,
+                       {"dynamic_reordering" : True}) as s:
+            s.add_assertion(self.big_tree)
+            self.assertTrue(s.solve())
+
+    @skipIfSolverNotAvailable("bdd")
+    def test_reordering_algorithms(self):
+        from pysmt.solvers.bdd import BddSolver, BddOptions
+        for algo in BddOptions.CUDD_ALL_REORDERING_ALGORITHMS:
+            with BddSolver(get_env(), pysmt.logics.BOOL,
+                           {"dynamic_reordering" : True,
+                            "reordering_algorithm" : algo}) as s:
+                s.add_assertion(self.big_tree)
+                self.assertTrue(s.solve())
+
+    @skipIfSolverNotAvailable("bdd")
+    def test_fixed_ordering(self):
+        from pysmt.solvers.bdd import BddSolver
+        with BddSolver(get_env(), pysmt.logics.BOOL,
+                       {"static_ordering" : [self.x, self.y]}) as s:
+            s.add_assertion(self.big_tree)
+            self.assertTrue(s.solve())
+
+    @skipIfSolverNotAvailable("bdd")
+    def test_initial_ordering(self):
+        from pysmt.solvers.bdd import BddSolver
+        with BddSolver(get_env(), pysmt.logics.BOOL,
+                       {"static_ordering" : [self.x, self.y],
+                        "dynamic_reordering" : True}) as s:
+            s.add_assertion(self.big_tree)
```

Comment:
How can we test that this is really working? This test needs to better capture the semantic of this operation.